File: typescript/packages/x402/src/signers/keyProvider.ts
/**
 * KeyProvider - abstraction for private key management.
 *
 * Any key management backend (agent-wallet, HSM, etc.)
 * can be used with x402 signers by implementing this interface.
 */

/** Minimal interface that a key management backend must satisfy. */
export interface KeyProvider {
  /** Return the TRON base58check address. */
  getAddress(): string;
  /** Return the hex-encoded private key (without 0x prefix). */
  getPrivateKey(): string;
}

File: python/x402/src/x402_tron/signers/key_provider.py
"""
KeyProvider Protocol - abstraction for private key management.

Any key management backend (agent-wallet, HSM, remote signer, etc.)
can be used with x402 signers by implementing this protocol.
"""

from typing import Protocol, runtime_checkable


@runtime_checkable
class KeyProvider(Protocol):
    """Minimal interface that a key management backend must satisfy.

    agent-wallet's TronProvider / FlashProvider can be adapted via
    ``TronProviderAdapter``.
    """

    def get_address(self) -> str:
        """Return the TRON base58check address."""
        ...

    def get_private_key(self) -> str:
        """Return the hex-encoded private key (without 0x prefix)."""
        ...
